### ISSUE TITLE
It is possible?

### ISSUE THREAD

**Issue by devlfm** [Corsaair/redtamarin#95]
Hi,

I've been wondering.. It is possible to implement the opensource skia? the same way flutter did... Basically creating native apps that use skia to render directly to gpu.. we could also create a "starling" transpiler that converts all stage3d methods/commands to skia methods/commands.. and then use feathers to UI/UX..

Skia is very mature and is used by chrome and supported by google..

Am I too off?

**Comment by zwetan** [comment]
It in itself yes you are a bit off because the mission of Redtamarin is to support the CLI.

> To support the use of the AS3 language for cross-platform command-line executable,
as single exe either for the desktop or the server, as scripts for automation,
as tools for the Flash Platform community.

It's not the goal of Redtamarin to do any GUI rendering.

Now, that said, you could combine Redtamarin with many different other things that does GUI rendering, for ex: QT, Chromium Embedded Framework (CEF), Skia, etc.

But that would be a separate project.

**Comment by rui-cruz** [comment]
Understood :)

It would be nice to see examples (videos or snapshoots)..

Let's see if I'm too off again:
I need to implement ImageMagik, actually I'm using a batch file with
NativeProcess.. but with redtamarin I could have ImageMagik sources
compiled and access it like a worker? Or.. redtamarin would just replace
the use of the "batch" file?



2017-04-07 10:58 GMT+01:00 Zwetan Kjukov <notifications@github.com>:

**Comment by zwetan** [comment]
that's a different problem
eg. rendering images vs manipulating images

but yeah so far to reuse ImageMagik you could use a class `NativeProcess`
in fact in next release 0.4.2 under Linux/MacOS you could simply use
`popen2()` and/or `popen3()` to do that

So, yes Redtamarin could include and compile against ImageMagik,
and provide native API to access ImageMagik functions, constants, etc.
and proba ly reuse those to implement things like `Bitmap`, `BitmapData`, etc.

Now, the problem would be to decide if we go with ImageMagik or something else

For example, you could implement as3corelib [PNGEncoder.as](https://github.com/mikechambers/as3corelib/blob/master/src/com/adobe/images/PNGEncoder.as) without `Bitmap` or `BitmapData`, and so manipulate PNG files

**Comment by devlfm** [comment]
Basically, due limitations of Flash API, I'm rendering content with
commands.. and running on a server I have no problems with that.. but I'm
thinking on move that rendering process the customer machine..

I believe I can archive this also with crossbridge.io no? and create a
FREObjects to comunicate with Flash..

Unfortunately my knowledge of c/c++ is very poor... being able to compile
its no guarantee that its well done..

2017-04-07 11:26 GMT+01:00 Zwetan Kjukov <notifications@github.com>:

**Comment by zwetan** [comment]
I don't understand what you are trying to do and why you are trying to do it this way ...

As far as Redtamarin is concerned, from your original question to this last post
I'm not sure how things are related to each other.

Maybe try to explain in greater details what/why you are trying to do here
https://discuss.as3lang.org/

**Comment by rui-cruz** [comment]
Well.. I'm definitely off... lol

What I truly truly want, is a skia wrapper for each platform generated by
falcon or other abc interpreter.. then I code in as3/mxml (keeping IDE's /
codebase) and release to whatever I wan't.. always rendering on gpu and
completely replacing flashplayer/air but with the same trust on rendered
stuff that flashplayer delivers atm. Basically what google is doing with
dart/flutter.. but I'll stop dreaming now.. lol

I guess you misunderstood my needs because I even know what I really need..
I guess I'm going through some type of menopause on flash.

I thought this project could be a starting point for that, because already
interprets as3 code, then something could be done to apply that code to a
rendering engine..

Then when you said that the main point of redtamarin is CLI, and I moved to
another question/subject.. but in the end, I'm just a frontend developer
and all this is too much for me :D

Sorry for the trouble :)


2017-04-07 18:37 GMT+01:00 Zwetan Kjukov <notifications@github.com>:

**Comment by zwetan** [comment]
yeah well the problem is you want everything and anything

eg. a kind of open source AIR based on Skia but that can also be cross-compiled with Falcon/FlexJS so it can generate things on HTML targets ...

in short, you want a "magic button" that publish to anywhere

again, that's not the goal of Redtamarin
we focus on the CLI to complement other runtimes like Flash/AIR where you can do GUI
it's not at all the goal of Redtamarin to replace Flash and/or AIR
it's not the goal to make an open source Flash-like player

simply put, its not impossible
and Redtamarin could be a good "first piece of the puzzle" to do that
but it need a lot of resources: time, developers actually writing code
not only C/C++ code but also AS3 code

If you want AS3 code that exports to HTML/JS look into FlexJS
for the rest, AS3 code can publish to mobile and desktop with Adobe AIR
as long as this is working I don't see a need to re-invent it

**Comment by devlfm** [comment]
Fair enough :)

Thank you

2017-04-07 20:37 GMT+01:00 Zwetan Kjukov <notifications@github.com>:
